### PR TITLE
feat (HACBS-1139): copy annotations/labels to other objs

### PR DESCRIPTION
**Pull request by jinqi7**
Signed-off-by: Jing Qi <jinqi@redhat.com>

### CODE REVIEW

File: controllers/snapshot/snapshot_adapter.go [hunk 1]
```diff
@@ -285,6 +286,24 @@ func (a *Adapter) createMissingReleasesForReleasePlans(releasePlans *[]releasev1
 				"Application.Name", a.application.Name,
 				"ReleasePlan.Name", releasePlan.Name,
 				"Release.Name", newRelease.Name)
+			//Copy PipelineRun PAC annotations/labels from applicaitonSnapshot to Release
```
Comment (dirgim) [inline controllers/snapshot/snapshot_adapter.go:289]: The copying of labels and annotations should happen before the `a.Client.Create(a.context, newRelease)` call on line 281 in order for them to be present on the created object.

File: controllers/snapshot/snapshot_adapter.go [hunk 2]
```diff
@@ -356,6 +375,23 @@ func (a *Adapter) createIntegrationPipelineRun(application *hasv1alpha1.Applicat
 	if err != nil {
 		return err
 	}
+	//Copy PipelineRun PAC annotations/labels from applicaitonSnapshot to integration test PipelineRuns
```
Comment (dirgim) [inline controllers/snapshot/snapshot_adapter.go:378]: Similar to the above, copying of labels and annotations should happen before the `a.Client.Create(a.context, newRelease)` call on line 374 in order for them to be present on the created object.

File: controllers/snapshot/snapshot_adapter.go (hunk 1 above)
Comment (jinqi7) [reply controllers/snapshot/snapshot_adapter.go:281]: Updated.

File: controllers/snapshot/snapshot_adapter.go (hunk 2 above)
Comment (jinqi7) [reply controllers/snapshot/snapshot_adapter.go:373]: Updated

File: controllers/snapshot/snapshot_adapter.go [hunk 3]
```diff
@@ -351,6 +370,23 @@ func (a *Adapter) createIntegrationPipelineRun(application *hasv1alpha1.Applicat
...
+	if applicationSnapshot.Annotations != nil {
+		pipelineRun.ObjectMeta.Annotations = make(map[string]string)
+		for key, value := range applicationSnapshot.Annotations {
+			if strings.Contains(key, "pipelinesascode.tekton.dev") {
+				pipelineRun.ObjectMeta.Annotations[key] = value
+			}
+		}
+	}
+	if applicationSnapshot.Labels != nil {
+		pipelineRun.ObjectMeta.Labels = make(map[string]string)
```
Comment (dirgim) [inline controllers/snapshot/snapshot_adapter.go:383]: This line will erase all labels that were already set on the `pipelineRun` object. I'd say we want to do the `make` call only if the `pipelineRun.ObjectMeta.Labels` are `nil`.

Comment (jinqi7) [reply controllers/snapshot/snapshot_adapter.go:383]: Yes. It need to be removed.

**Review by dirgim — APPROVED** [review]
LGTM

File: controllers/snapshot/snapshot_adapter.go (hunk 3 above)
Comment (dirgim) [reply controllers/snapshot/snapshot_adapter.go]: Yes, looks good now, thanks!

**Review by jsztuka — APPROVED** [review]
lgtm

**Pull request merged**